s4cmb/tod.py

```diff
@@ -772,19 +870,30 @@ def map2tod(self, ch):
 
         ## Compute pointing for detector ch
         ra, dec, pa = self.pointing.offset_detector(azd, eld)
-
+        
+        # Perturbed boresight pointing values
```

Comment:
It's useless to compute the perturbed pointing every time. Following my previous suggestion

`if self.pointing_perturbed is not None:
        ra_perturbed, dec_perturbed, pa_perturbed = self.pointing_perturbed.offset_detector(azd, eld)
else:
         ra_perturbed = ra
         dec_perturbed = dec
         pa_perturbed = pa`